### PR TITLE
Network - v3 - fixed all the appointments

### PR DESCRIPTION
**Pull request by amandawrcoelho**
Changed all the -errno returns.

issue #1716

Signed-off-by: Amanda Coelho <amada.winder.ribeiro.coelho@intel.com>

### CODE REVIEW

File: src/lib/comms/sol-network-impl-linux.c [hunk 1]
```diff
@@ -574,19 +577,19 @@ sol_network_link_set_status(uint16_t link_index, unsigned int changes, unsigned
 
     if (sendmsg(network->nl_socket, (struct msghdr *)&msg, 0) <= 0) {
         SOL_WRN("Failed on send message to set link up");
-        return false;
+        return -errno;
```
Comment (amandawrcoelho) [inline src/lib/comms/sol-network-impl-linux.c]: Change it

File: src/lib/comms/sol-network-impl-linux.c [hunk 2]
```diff
@@ -455,42 +455,45 @@ sol_network_shutdown(void)
...
 {
+    int r;
+
     struct callback *callback;
 
-    SOL_NULL_CHECK(cb, false);
+    SOL_NULL_CHECK(cb, -EINVAL);
 
-    if (sol_network_start_netlink() < 0) return false;
+    r = sol_network_start_netlink;
```
Comment (cabelitos) [inline src/lib/comms/sol-network-impl-linux.c]: Did you compile this? I guess you forgot to add `()`

File: src/lib/comms/sol-network-impl-linux.c [hunk 3]
```diff
@@ -532,7 +535,7 @@ sol_network_link_set_status(uint16_t link_index, unsigned int changes, unsigned
     uint8_t mode = IN6_ADDR_GEN_MODE_EUI64;
     size_t buf_size = sizeof(buf);;
 
-    if (sol_network_start_netlink() < 0) return false;
+    if (sol_network_start_netlink() < 0) return -ENOMEM;
```
Comment (cabelitos) [inline src/lib/comms/sol-network-impl-linux.c]: No no, you can't simple return `-ENOMEM` here. The function `sol_network_start_netlink()` already returns the error value. You should do the following:

```
int r;
...
r = sol_network_start_netlink();
SOL_INT_CHECK(r, < 0, r);
```

File: src/lib/comms/sol-network-impl-linux.c [hunk 4]
```diff
@@ -353,7 +353,7 @@ sol_network_start_netlink(void)
     network->nl_socket = socket(AF_NETLINK, SOCK_RAW | SOCK_CLOEXEC | SOCK_NONBLOCK, NETLINK_ROUTE);
     if (network->nl_socket < 0) {
         SOL_WRN("failed to create netlink socket, cannot listen to events or manage the links!");
-        return -errno;
+        return -EBADMSG;
```
Comment (cabelitos) [inline src/lib/comms/sol-network-impl-linux.c]: In this case returning the `-errno` is valid. The `socket()` will set the errno with the correctly value if something goes wrong

Comment (cabelitos) [inline src/lib/comms/sol-network-impl-linux.c]: Remenber to check the man pages.

File: src/lib/comms/sol-network-impl-linux.c (hunk 1 above)
Comment (cabelitos) [inline src/lib/comms/sol-network-impl-linux.c]: What do you mean?

Comment (ibriano) [inline src/lib/comms/sol-network-impl-linux.c]: Careful with those, even the if the function uses errno to set the error, SOL_WRN (and related) may re-set errno to some other value. In these cases you need to store errno to a temporary variable before you call anything else, and then return (or set back) that value.

Comment (cabelitos) [inline src/lib/comms/sol-network-impl-linux.c]: I checked. SOL_WRN will not mess with the errno value

**Pull request closed without merging**